client/my-sites/themes/controller.jsx

```diff
@@ -99,20 +97,23 @@ export function fetchThemeFilters( context, next ) {
 		return next();
 	}
 
-	const unsubscribe = store.subscribe( () => {
-		const fetchErr = getThemeFiltersRequestError( store.getState() );
-		if ( fetchErr ) {
-			debug( `Theme fetch error: ${ JSON.stringify( fetchErr ) }` );
-			unsubscribe();
-			return next( new Error( 'Error fetching theme filters.' ) );
-		}
-
-		if ( Object.keys( getThemeFilters( store.getState() ) ).length > 0 ) {
-			unsubscribe();
-			return next();
-		}
+	const filtersRequest = wpcom.req.get( '/theme-filters', {
+		apiVersion: '1.2',
+		locale: context.lang, // Note: undefined will be omitted by the query string builder.
 	} );
-	store.dispatch( requestThemeFilters( context.lang ) );
+
+	const timeout = new Promise( ( _, reject ) => {
+		setTimeout( () => {
+			reject( new Error( 'Theme filters request timed out' ) );
```

Comment:
Can the `/theme-filters` API still actually timeout, and would it be worth adding Tracking here to see if this error gets fired?

Assuming it would show up as non-functional theme filters to customers?

Reply:
It definitely could -- this error should actually be logged automatically by the `next` middleware

Reply:
However, it shouldn't impact customers too much. When `next` is called with an error, it should bypass the SSR renderer, and basically just let the client render the page instead, at which point the client will fetch that data

Reply:
(every error passed to `next` should be automatically logged to logstash)